src/NodeHelper.groovy

```diff
@@ -0,0 +1,1136 @@
+
+        if (value.length() > 1) {
+            switch (value) {
+                case "amd64":
+                    ret = "x";
+                    break;
+                case "ppc64":
+                case "ppc64le":
+                    ret = "p";
+                    break;
+                case "s390x":
+                    ret = "z";
+                case "arm":
+                    ret = "arm";
+                    break;
+            }
+        }
+
+        return ret;
+    }
+
+    public String constructArchLabel(String computerName) {
+        String ret = "constructArchLabel:COMPUTER_NOT_FOUND";
+
+        Computer computer = getComputer(computerName);
+        if (computer != null) {
+            String arch = getOsArch(computer.getName());
+
+            switch (arch) {
+                case "x86_64":
+                case "amd64":
+                    ret = "x86";
+                    break;
+                case "ppc64":
+                case "ppc64le":
+                    ret = "ppc";
+                    break;
+                case "s390x":
+                    ret = "s390";
+                    break;
+                case "arm":
+                    ret = arch;
+                    break;
+                default:
+                    ret = "INVALID_ARCH";
+                    break;
+            }
+            ret = "hw.arch." + ret;
+        }
+
+
+        return ret.toLowerCase();
+    }
+
+    public String constructKernelLabel(String computerName) {
+        String ret = "constructKernelLabel:COMPUTER_NOT_FOUND";
+
+        Computer computer = getComputer(computerName);
+        if (computer != null) {
+            def kernelInfo = getOsKernelInfo(computer.getName());
+            ret = "sw.os." + kernelInfo.get(0);
+        }
+
+        return ret;
+    }
+
+    public String constructHypervisorLabel(String computerName) {
+        String ret = "constructHypervisorLabel:COMPUTER_NOT_FOUND";
+
+        Computer computer = getComputer(computerName);
+        if (computer != null) {
+            ret = "hw.hypervisor.";// TODO: finish implementation, get something
+        }
+
+        return ret.toLowerCase();
+    }
+
+    
+    public Tuple getOsInfo(String computerName) {
+        Tuple ret;
+
+        Computer computer = getComputer(computerName);
+        if (computer != null) {
+            def osInfo
+            switch (getOsKernelInfo(computer.getName()).get(0)) {
+                case "linux":
+                    ret = getLinuxOsInfo(computer);
+                    break;
+                case "windows":
+                    ret = getWindowsOsInfo(computer);
+                    break;
+                case "aix":
+                    ret = getAIXOsInfo(computer);
+                    break;
+                case "zos":
+                    // ret = // TODO: find the command
+                    // break; 
+                case "mac":
+                    // ret = getMacOsInfo(computer);
```

Comment:
commented out on purpose?

Reply:
Yes, wasn't sure if we needed the version number since it is the same as kernel version. I implement it and check it in with the next commit